frontend/src/app/shared/components/file-links/file-picker-modal/file-picker-modal.component.ts

```diff
@@ -148,6 +149,21 @@ export class FilePickerModalComponent extends OpModalComponent implements OnInit
     this.service.close();
   }
 
+  public selectAllOfCurrentLevel():void {
+    this.storageFiles$
+      .pipe(take(1))
+      .subscribe((files) => {
+        files.forEach((file) => {
+          const id = file.id as string;
+          if (!this.selection.has(id) && !this.isAlreadyLinked(file)) {
+            this.selection.add(id);
+          }
+        });
+
+        this.storageFiles$.next(files);
```

Comment:
is this necessary? If you want to force a rerender, a manual change detection run might be more clear.

Reply:
I will give it a try, but I found it easier to just repush it to the subject. because I do not have to inject the `cdRef`.

Reply:
Yeah just because of that.

Reply:
I tried it, and just replacing it with change detection, doesn't work. The reason therefor is line 115ff, the subscriber that builds the list item models out of the file data is not triggered by a cd cycle.